Gem sim validation porting 76 x only occupy plots

geonmo opened the pull request:
### Hit Validation module for GEM Detector after Run2
1. This PR include Validation/MuonGEMHits, MuonGEMDigis, MuonGEMRecHits module to validate GEM hit simulation.
   - MuonGEMHits module is related to SimHit on GEM Detector(Run2019/2023).
   - MuonGEMDigis module is related to Digi step about GEM Strip, pad and Copad.
   - MuonGEMRechits module is related to local reco hit on GEM Detector. 
#### This PR does not affect to running other modules.
- These module are only running with specific customize options.

Remake PR due to wrong push to my branch and too old.
It is same PR #11123 except some fix as follow comments.

geonmo committed:
Add MuonGEM Simhit, digiHit and recoHit occupancy plots

geonmo committed:
Merged gem-sim-validation-porting-76X-only_occupyPlots from repository geonmo

geonmo committed:
Fix some typo when removed efficiency fetecture.

geonmo committed:
Merged gem-sim-validation-porting-76X-only_occupyPlots from repository geonmo

geonmo committed:
Remove efficiency plots before removing noise.

Commit:
Add deprecated warning to postLS1 customisation functions

Commit:
Change from customisePostLS1 customisations to Run2 eras. Premixing left untouched

mulhearn committed:
initial commit of 2016 L1T muon emulation

Commit:
added 3D and beamspot/vertex options

Commit:
Switch on hadronFlavourHasPriority for b-tag Validation module.

geonmo committed:
Merged gem-sim-validation-porting-76X-only_occupyPlots from repository geonmo

fioriNTU committed:
Make FED Occupancy plot a TProfile

geonmo committed:
Refactoring and bugfix as comments.

cmsbuild commented:
A new Pull Request was created by @geonmo (Geonmo Ryu) for CMSSW_7_6_X.

Gem sim validation porting 76 x only occupy plots

It involves the following packages:

Validation/MuonGEMDigis
Validation/MuonGEMHits
Validation/MuonGEMRecHits


@cmsbuild, @danduggan, @vanbesien, @deguio can you please review it and eventually sign? Thanks.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

geonmo committed:
Rename cfi to cff.

cmsbuild commented:
Pull request #11605 was updated. @cmsbuild, @danduggan, @vanbesien, @deguio can you please check and sign again.

cmkuo committed:
fix ES unpacker when the 2nd OptoRX is diabled, but the 1t and 3rd OptoRX are not

geonmo commented:
@deguio Hello, Federico. I re-make a PR due to mistake on previous PR. Could you test this?

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: e257896c2dafa71bb0b13a2c7d519f6f38489ccd
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11605/8580/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11605/8580/summary.html

Commit:
Make FWEventItem::getPrimaryData public

`FWEventItem::getPrimaryData` is being used by `FWFileEntry` (Fireworks
code), but uses `#define private public` to get access to it. The patch
makes it public that private re-define could be removed.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Fireworks: move to public ROOT APIs

Fireworks code relies a lot on re-defining {private,protected} to get
access to various ROOT classes internals. Those re-defines are also
breaking GNU C++ Standard Library (libstdc++v3). Some are already
available via ROOT public APIs and the patch updates Fireworks code to
utilise them.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

smuzaffar committed:
set reco::CandViewCandViewAssociation as transient

fioriNTU committed:
restore correct histo binning

Commit:
put some includes back in the .cc file

Commit:
calculate angles using dot product

geonmo commented:
@deguio Hello, Federico. Could you review for this PR?

Commit:
Replacing SpikeNoise flag by NegativeNoise one

davidlange6 committed:
make hcal digitization changes the default

Commit:
use beamspot instead of beamspot position

Dr15Jones committed:
Fix precedence error and added additional functions

Further tests of expressions from the data base showed that there
were additional ones that couldn't be handled.

davidlange6 committed:
Merge pull request #11675 from davidlange6/hcalNewDefault

make hcal digitization changes the default

davidlange6 committed:
Merge pull request #11684 from Dr15Jones/bugFixFormulaEvaluator

Fix precedence error and added additional functions

davidlange6 committed:
Merge pull request #11647 from smuzaffar/from-CMSSW_7_6_X_2015-10-05-2300

set reco::CandViewCandViewAssociation as transient

davidlange6 committed:
Merge pull request #11572 from cms-btv-pog/fixBtagDQMFlavour

Switch on hadronFlavourHasPriority for b-tag Validation module.

davidlange6 committed:
Merge pull request #11504 from cms-l1t-offline/pr-l1t-muon-init-76x

initial commit of 2016 L1T muon emulation

cmsbuild committed:
Merge pull request #11614 from cmkuo/mydev_esunpacker_76x

fix ES unpacker when the 2nd OptoRX is diabled, but the 1t and 3rd Op…

davidlange6 committed:
Merge pull request #11601 from fioriNTU/FedOccProfile76x

Make FED Occupancy plot a TProfile

davidlange6 committed:
Merge pull request #11641 from davidlt/fireworks-cleanup-prep

Fireworks cleanup for {private,protected} macro removal

davidlange6 committed:
put the new time slew in by default

davidlange6 committed:
Merge pull request #11686 from davidlange6/newhcalHLT

put the new time slew in by default for HLT mc

deguio commented:
+1
detailed presentation at:
https://indico.cern.ch/event/446270/contribution/2/attachments/1154267/1658925/Presentation_on_DQM_MeetingGEM_Validation_modules.pdf

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_6_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

davidlange6 commented on Validation/MuonGEMDigis/interface/GEMCheckGeometry.h:
```diff
@@ -0,0 +1,43 @@
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/GEMDigi/interface/GEMDigiCollection.h"
+#include <TMath.h>
```
most of these header files are either not needed or should be in the c++.. could you try to optimize this a bit?

cmkuo committed:
take into account comments after last PR

cmkuo committed:
take into account comments after last PR

Commit:
New GT with PF hadron calibration.

davidlange6 committed:
Merge pull request #11466 from mark-grimes/switchToErasAndWarn

Switch runTheMatrix tests to use Run 2 eras

davidlange6 committed:
Merge pull request #11697 from diguida/autoCond_76X_PFCalibration

New GT with PF hadron calibration.

wmtan committed:
Replace cryptic assert with informative exceptions

davidlange6 committed:
Merge pull request #11699 from fojensen/V0forHLT

add 75x V0Producer to 76x

davidlange6 committed:
Merge pull request #11693 from cmkuo/mydev_esunpacker_76x

improve ES unpacker

cmsbuild committed:
Merge pull request #11703 from wmtan/ReplaceCrypticAssert

Replace cryptic assert with informative exception for missing Wrapper dictionary

davidlange6 committed:
Merge pull request #11707 from mark-grimes/fixL1StepEra

Add caloConfig and caloConfigSource to L1 step

davidlange6 committed:
Merge pull request #11667 from abdoulline/NEF_activation_76X

Replacing SpikeNoise flag by NegativeNoise one

geonmo committed:
Merge branch 'gem-sim-validation-porting-76X-only_occupyPlots' of github.com:geonmo/cmssw into from-CMSSW_7_6_0_pre7

geonmo committed:
Optimize the header file's include list.

geonmo commented:
@davidlange6 Hello, david! I am sorry about late updates. I am so busy to prepare our regional conference. Could you review my code about head optimization?

cmsbuild commented:
Pull request #11605 was updated. @diguida, @cerminar, @fabozzi, @Martin-Grunewald, @vanbesien, @perrotta, @civanch, @srimanob, @cmsbuild, @fwyzard, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @deguio, @slava77, @vadler, @mmusich, @mulhearn, @danduggan, @boudoul, @monttj, @alja, @franzoni can you please check and sign again.

geonmo commented:
Well, I re-checked a PR before push using "Comparision between base and head".
I think that it is not correctly working as my guess
Ok, I will remake a new PR with same contents.

geonmo commented:
@cmsbuild, @cmsbuild, @danduggan, @vanbesien, @deguio and @davidlange6 
Hello, everyone. I remake a new PR with same contents. Please, see PR#11948 for this issue.
Thanks
